enzy_htp/mutation/api.py

```diff
@@ -367,8 +369,59 @@ def mutate_stru_with_tleap(
 
     return stru_cpy
 
+def mutate_stru_with_pymol(
+        stru: Structure,
+        mutant: List[Mutation],
+        in_place: bool = False,
+        ) -> Structure:
+    """mutate the {stru} to its {mutant} structure using PyMOL.
+    Args:
+        stru: the 'WT' structure
+        mutant: a list of Mutation() which describes a mutant to the 'WT'
+        in_place: if make the changes to the structure in-place 
+    Returns:
+        The mutated structure.
+    """
+
+    sp = PDBParser()
+    temp_file_name = fs.get_valid_temp_name("pymol_mutation_result.pdb")
+
+    # san check
+    for mut in mutant:
+        if not isinstance(mut, Mutation):
+            _LOGGER.error(
+                f"mutant takes only a list of Mutation(). Current mutant is: {mutant}")
+            raise TypeError
+
+    stru_cpy = copy.deepcopy(stru)
+
+    # 1. load stru into pymol
+    pi = interface.pymol
+    temp_session = pi.new_pymol_session()
```

Comment:
The pymol_session is not closed. Please reference [this function](https://github.com/ChemBioHTP/EnzyHTP/blob/627bde52666d995680f5fd9f4580fc1e1e4179cf/enzy_htp/structure/structure_selection/general.py#L11) and use the context manager to handle the pymol session. (the design is like file IO)